### PR TITLE
ScanForPreImage Enhancement: Custom Starting Point #378

### PR DESCRIPTION
**Pull request by adecaro**
Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: integration/token/interop/tests.go [hunk 1]
```diff
@@ -149,18 +149,24 @@ func TestHTLCNoCrossClaimTwoFabricNetworks(network *integration.Infrastructure)
...
-	preImage, hash := htlcLock(network, alpha, "alice", "alice.id1", "EUR", 10, "alice.id2", 30*time.Second, nil, 0)
-	htlcLock(network, beta, "bob", "bob.id1", "USD", 10, "bob.id2", 30*time.Second, hash, 0)
+	aliceLockTxID, preImage, hash := htlcLock(network, alpha, "alice", "alice.id1", "EUR", 10, "alice.id2", 30*time.Second, nil, 0)
+	bobLockTxID, _, _ := htlcLock(network, beta, "bob", "bob.id1", "USD", 10, "bob.id2", 30*time.Second, hash, 0)
 
 	go func() { htlcClaim(network, alpha, "alice", "alice.id2", preImage) }()
 	go func() { htlcClaim(network, beta, "bob", "bob.id2", preImage) }()
-	scan(network, "alice", hash, crypto.SHA256, token.WithTMSID(alpha))
-	scan(network, "bob", hash, crypto.SHA256, token.WithTMSID(beta))
+	scan(network, "alice", hash, crypto.SHA256, "", token.WithTMSID(alpha))
```
Comment (HagarMeir) [inline integration/token/interop/tests.go:157]: Why are we even using scan in this test?

**Comment by HagarMeir** [comment]
Is there a way to make this `ScanForPreImage` backed agnostic by using the Network API (token/services/network)?
This was one of your suggestions for enhancements.

File: integration/token/interop/tests.go (hunk 1 above)
Comment (adecaro) [reply integration/token/interop/tests.go:157]: just to check the different type of scan works as expected.

**Comment by adecaro** [comment]
Yeah, I think there is. I will address in another PR. Here, I just wanted to address the starting point. Let me change the title to better reflect

**Review by HagarMeir — APPROVED** [review]

**Pull request merged**